Begin generating IPC serialization code

achristensen07 opened the pull request:
#### 11ca86229cda5d387fc45c3418d5a59b16261bf7
<pre>
Begin generating IPC serialization code
<a href="https://bugs.webkit.org/show_bug.cgi?id=244402">https://bugs.webkit.org/show_bug.cgi?id=244402</a>

Reviewed by Chris Dumez.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py: Added.
(SerializedType):
(SerializedType.__init__):
(SerializedType.namespace_and_name):
(MemberVariable):
(MemberVariable.__init__):
(generate_header):
(generate_header.Decoder):
(generate_header.Encoder):
(generate_header.StreamConnectionEncoder):
(generate_cpp):
(parse_serialized_type):
(main):
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_handler):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp: Added.
(IPC::ArgumentCoder&lt;Namespace::Subnamespace::StructName&gt;::encode):
(IPC::ArgumentCoder&lt;Namespace::Subnamespace::StructName&gt;::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h: Added.
* Source/WebKit/Scripts/webkit/tests/Makefile:
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in: Added.
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.h:
(WebKit::WebGPU::Origin3DDict::encode const): Deleted.
(WebKit::WebGPU::Origin3DDict::decode): Deleted.
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.serialization.in: Added.
* Source/WebKit/Shared/WebGPU/WebGPUValidationError.h:
(WebKit::WebGPU::ValidationError::encode const): Deleted.
(WebKit::WebGPU::ValidationError::decode): Deleted.
* Source/WebKit/Shared/WebGPU/WebGPUValidationError.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:

Canonical link: <a href="https://commits.webkit.org/253920@main">https://commits.webkit.org/253920@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/27e4eceb402db0d40332a15ab121a0471d65618f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87446 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31535 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18252 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96674 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29900 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/26084 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/79565 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91445 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/93064 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/24160 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74232 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23998 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/91048 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/79127 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79352 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/67016 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/79289 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27610 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13184 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/72970 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27564 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14200 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/25984 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2752 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29252 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/37060 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/75796 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/29179 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33478 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/16800 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/DerivedSources.make:
```diff
@@ -445,3 +445,13 @@ $(WEB_PREFERENCES_COMBINED_INPUT_FILE) : $(WEB_PREFERENCES_INPUT_FILES)
+SERIALIZATION_DESCRIPTION_FILES = \
+	$(WebKit2)/Shared/WebGPU/WebGPUValidationError.serialization.in \
+	$(WebKit2)/Shared/WebGPU/WebGPUOrigin3D.serialization.in \
+#
```
Is this intentional?

achristensen07 replied:
Yes.  That's so we can have a \ at the end of the last line and keep the revision history cleaner.  That's how other lists are done in this file.

cdumez reviewed (approved):
LGTM

achristensen07 committed:
Begin generating IPC serialization code
https://bugs.webkit.org/show_bug.cgi?id=244402

Reviewed by Chris Dumez.

* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py: Added.
(SerializedType):
(SerializedType.__init__):
(SerializedType.namespace_and_name):
(MemberVariable):
(MemberVariable.__init__):
(generate_header):
(generate_header.Decoder):
(generate_header.Encoder):
(generate_header.StreamConnectionEncoder):
(generate_cpp):
(parse_serialized_type):
(main):
* Source/WebKit/Scripts/webkit/messages.py:
(generate_message_handler):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.cpp: Added.
(IPC::ArgumentCoder<Namespace::Subnamespace::StructName>::encode):
(IPC::ArgumentCoder<Namespace::Subnamespace::StructName>::decode):
* Source/WebKit/Scripts/webkit/tests/GeneratedSerializers.h: Added.
* Source/WebKit/Scripts/webkit/tests/Makefile:
* Source/WebKit/Scripts/webkit/tests/TestSerializedType.serialization.in: Added.
* Source/WebKit/Scripts/webkit/tests/TestWithCVPixelBufferMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithIfMessageMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithImageDataMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithLegacyReceiverMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithSemaphoreMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBatchedMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamBufferMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithStreamMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithSuperclassMessageReceiver.cpp:
* Source/WebKit/Scripts/webkit/tests/TestWithoutAttributesMessageReceiver.cpp:
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.h:
(WebKit::WebGPU::Origin3DDict::encode const): Deleted.
(WebKit::WebGPU::Origin3DDict::decode): Deleted.
* Source/WebKit/Shared/WebGPU/WebGPUOrigin3D.serialization.in: Added.
* Source/WebKit/Shared/WebGPU/WebGPUValidationError.h:
(WebKit::WebGPU::ValidationError::encode const): Deleted.
(WebKit::WebGPU::ValidationError::decode): Deleted.
* Source/WebKit/Shared/WebGPU/WebGPUValidationError.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteDeviceProxy.cpp:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteQueueProxy.cpp:

Canonical link: https://commits.webkit.org/253920@main
